elastic/elasticsearch.go

```diff
@@ -0,0 +1,121 @@
+package elasticsearch
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/olivere/elastic"
+	aws "github.com/olivere/elastic/aws/v4"
+)
+
+// ElasticClient wraps a Client for Elasticsearch interactions. The struct also includes a logger, that will get configured, in instantiation methods.
+type ElasticClient struct {
+	Client *elastic.Client
+	Logger *log.Logger
+}
+
+// ElasticAWSConfig wraps necessary configuration for AWS Elasticsearch Client.
+type ElasticAWSConfig struct {
+	Region      string
+	URL         string
+	AccessKey   string
+	SecretKey   string
+	ServiceName string
+}
+
+// ElasticQueryResult is a simple wrapper implementation of the results of an ElasticQuery.
+type ElasticQueryResult struct {
+	RecordIDs []string
```

Comment:
comments for these three fields would be useful to non elastic search experienced people( I think lines 23-27 are pretty self evident)